addons/mail/models/mail_render_mixin.py

```diff
@@ -203,15 +202,16 @@ def _prepend_preview(self, html, preview):
         if preview:
             preview = preview.strip()
 
+        preview_markup = convert_inline_template_to_qweb(preview)
+
         if preview:
             html_preview = Markup("""
                 <div style="display:none;font-size:1px;height:0px;width:0px;opacity:0;">
-                   {}
+                    {}
                 </div>
-            """).format(preview)
-            return tools.prepend_html_content(html, html_preview)
+            """).format(preview_markup)
+            return tools.prepend_html_content(Markup(html), html_preview)
```

Comment:
I'm not sure this is necessary, the only call to `_prepend_preview` I can see where it might not be passed a `Markup` for sure would be `send_mail_test`, in which case that is where the marking should be applied. But even then it might already be marked: `_render_field` returns the output of `_render_template`, whose outputs should be marked implicitely for the two qweb case, was so explicitly for the jinja case, and since its replacement should mark the output correctly via `render_inline_template` it should be fine?